src/lib.rs

```diff
@@ -180,6 +185,14 @@ impl Repr {
         }
     }
 
+    fn is_empty(&self) -> bool {
+        match self {
+            Repr::Heap(data) => data.is_empty(),
+            Repr::Inline { len, .. } => *len == 0,
+            Repr::Substring { newlines, spaces } => (*newlines + *spaces) == 0,
```

Comment:
this can be `False` I think: we gurantee that an empty string uses `Inline` repr.